async-coap/src/datagram/allow_udp_socket.rs

```diff
@@ -91,13 +90,9 @@ impl AllowStdUdpSocket {
         if let Some(d) = self.2 {
             let mut lock = self.1.lock().expect("Lock failed");
             let opt_mut: &mut Option<Delay> = &mut lock;
-            if opt_mut.is_none() {
-                *opt_mut = Some(Delay::new(d));
-                delay = opt_mut.as_mut().unwrap();
-            } else {
-                delay = opt_mut.as_mut().unwrap();
-                delay.reset(d);
-            }
+
+            *opt_mut = Some(Delay::new(d));
+            delay = opt_mut.as_mut().unwrap();
```

Comment:
This change doesn't immediately look related to the futures update. Can you elaborate?

Reply:
hm.. i was almost 100% sure, that Delay::reset has been removed from the API, at some point. I will check and if it is still there, obviously, this change does not make sense, and I will revert it.

Reply:
just pushed the commit, which reverses unneeded Delay re-creation. Although, some small change is still needed - Delay::reset chnaged it's signature.